TestFoundation/TestNSFileManager.swift

```diff
@@ -239,4 +241,94 @@ class TestNSFileManger : XCTestCase {
         }
     }
     
+    func test_contentsOfDirectoryAtPath() {
+        let fm = NSFileManager.defaultManager()
+        let path = "/tmp/testdir"
+        let itemPath1 = "/tmp/testdir/item"
+        let itemPath2 = "/tmp/testdir/item2"
+        
+        let expected: [String] = ["item", "item2"]
+        
+        ignoreError { try fm.removeItemAtPath(path) }
+        
+        do {
+            try fm.createDirectoryAtPath(path, withIntermediateDirectories: false, attributes: nil)
+            fm.createFileAtPath(itemPath1, contents: NSData(), attributes: nil)
+            fm.createFileAtPath(itemPath2, contents: NSData(), attributes: nil)
+        } catch _ {
+            XCTFail()
+        }
+        
+        do {
+            let entries = try fm.contentsOfDirectoryAtPath(path)
+            
+            XCTAssertEqual(entries, expected)
```

Comment:
this fails on ubuntu 14

["item2", "item"] should be ["item", "item2"]

Mac OS X returns the directory contents as sorted, ubuntu does not do this, so perhaps it should be an un-ordered collection comparison. e.g. verify the count and verify that the expected items are contained in the result array